File: bindings/nodejs/examples/basic.ts
/**
 * Basic usage example for QUAC 100 SDK
 */

import {
    initialize,
    cleanup,
    getVersion,
    getBuildInfo,
    getDeviceCount,
    enumerateDevices,
    openFirstDevice,
} from '../src';

async function main() {
    console.log('=== QUAC 100 SDK Basic Example ===\n');

    // Show version
    console.log(`Library Version: ${getVersion()}`);
    console.log(`Build Info: ${getBuildInfo()}\n`);

    // Initialize library
    console.log('Initializing library...');
    initialize();
    console.log('Library initialized.\n');

    // Enumerate devices
    const deviceCount = getDeviceCount();
    console.log(`Found ${deviceCount} device(s)\n`);

    if (deviceCount === 0) {
        console.log('No devices found. Please connect a QUAC 100 device.');
        cleanup();
        return;
    }

    // List all devices
    const devices = enumerateDevices();
    for (const info of devices) {
        console.log(`Device ${info.index}:`);
        console.log(`  Model: ${info.model}`);
        console.log(`  Serial: ${info.serialNumber}`);
        console.log(`  Firmware: ${info.firmwareVersion}`);
        console.log(`  Key Slots: ${info.keySlots}`);
        console.log();
    }

    // Open first device
    console.log('Opening first device...');
    const device = openFirstDevice();
    console.log('Device opened.\n');

    // Get device status
    const status = device.getStatus();
    console.log('Device Status:');
    console.log(`  Temperature: ${status.temperature}°C`);
    console.log(`  Entropy Level: ${status.entropyLevel}%`);
    console.log(`  Operations: ${status.operationsCount}`);
    console.log(`  Uptime: ${status.uptime}s`);
    console.log(`  Ready: ${status.isReady}`);
    console.log();

    // Run self-test
    console.log('Running self-test...');
    device.selfTest();
    console.log('Self-test passed!\n');

    // Clean up
    console.log('Closing device...');
    device.close();
    console.log('Device closed.\n');

    console.log('Cleaning up library...');
    cleanup();
    console.log('Done!');
}

main().catch(console.error);

File: bindings/rust/examples/basic.rs
//! Basic usage example for QUAC 100 SDK.
//!
//! This example demonstrates library initialization, device enumeration,
//! and basic operations.

use quantacore::{
    initialize, cleanup, get_version, get_device_count,
    enumerate_devices, open_first_device,
};

fn main() -> quantacore::Result<()> {
    println!("QUAC 100 SDK Basic Example");
    println!("==========================\n");

    // Print version
    println!("SDK Version: {}", get_version());
    println!();

    // Initialize the library
    println!("Initializing library...");
    initialize()?;
    println!("Library initialized successfully.\n");

    // Enumerate devices
    let device_count = get_device_count();
    println!("Found {} QUAC 100 device(s).\n", device_count);

    if device_count == 0 {
        println!("No devices found. Please ensure a QUAC 100 is connected.");
        cleanup()?;
        return Ok(());
    }

    // List all devices
    for device_info in enumerate_devices() {
        println!("Device {}:", device_info.index);
        println!("  Model: {}", device_info.model);
        println!("  Serial: {}", device_info.serial_number);
        println!("  Firmware: {}", device_info.firmware_version);
        println!("  Key Slots: {}", device_info.key_slots);
        println!();
    }

    // Open the first device
    println!("Opening first device...");
    let device = open_first_device()?;
    println!("Device opened successfully.\n");

    // Get device status
    let status = device.get_status()?;
    println!("Device Status:");
    println!("  Temperature: {}°C", status.temperature);
    println!("  Entropy Level: {}%", status.entropy_level);
    println!("  Operations: {}", status.operation_count);
    println!("  Uptime: {} seconds", status.uptime_seconds);
    println!("  Healthy: {}", status.is_healthy());
    println!();

    // Run self-test
    println!("Running self-test...");
    device.self_test()?;
    println!("Self-test passed.\n");

    // Close device and cleanup
    drop(device);
    cleanup()?;
    println!("Cleanup complete.");

    Ok(())
}